ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiParticipantPruningService.scala

```diff
@@ -36,18 +41,70 @@ final class ApiParticipantPruningService private (
     writeBackend: state.WriteParticipantPruningService,
     metrics: Metrics,
 )(implicit executionContext: ExecutionContext, loggingContext: LoggingContext)
-    extends ParticipantPruningServiceGrpc.ParticipantPruningService
+    extends SinglePruningOpEnforcer
```

Comment:
Here we enforce one pruning operation at a time. I've implemented it as a UX improvement, but am slightly questioning its usefulness now since:
* The backing stores (write service and LAPI state) allow concurrent pruning requests
* After a HA failover we would still be able to end up with two concurrent pruning ops from different participants.

Any thoughts?

Reply:
I would keep this around, since it covers the majority of the abuse cases. Just that we probably can do parallel iterative pruning, it does not mean it is health, performant, or would look good in logs etc.
Although I would also double check this with @oliverse-da so we are not accidentally breaking some API-call-idempotency with introducing errors.

Reply:
fwiw, canton participants queue up "concurrent prune requests" of read/write-service stores and don't process them concurrently. Automatic canton participant pruning will not issue concurrent requests, but a user may issue a concurrent "manual" prune request. The `SinglePruningOpEnforcer` should not cause an issue, but it may be worth invoking concurrent participant prune calls on the same participant from the canton console `participant1.pruning.prune` to ensure that the resulting error shows up the way we want.